pkgs/tools/system/fancy-motd/default.nix

```
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchFromGitHub, bc, curl, figlet, fortune, gawk, iproute2, procps }:
+
+stdenv.mkDerivation rec {
+  pname = "fancy-motd";
+  version = "20211305";
+
+  src = fetchFromGitHub {
+    owner = "bcyran";
+    repo = pname;
+    rev = "9c93a6d2058029aa82e231ccad3a9896ff413c3e";
+    sha256 = "16v6j2nwgj1njzk91vd9bqjlh8hj30wadxzpy7dwlf816wnh4cgh";
+  };
+
+  buildInputs = [ bc curl figlet fortune gawk iproute2 ];
+
+  postPatch = ''
+    substituteInPlace motd.sh \
+      --replace 'BASE_DIR="$(dirname "$(readlink -f "$0")")"' "BASE_DIR=\"$out/lib\""
+
+    substituteInPlace modules/20-uptime \
+      --replace "uptime -p" "${procps}/bin/uptime -p"
+
+    # does not work on nixos
+    rm modules/41-updates
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp motd.sh $out/bin/motd
```

```suggestion
    install -D motd.sh $out/bin/motd
```

same for the later mkdir && copy pairs